packages/scripts/src/admin-proposals/register-parent-messenger/0_Propose.ts

```diff
@@ -0,0 +1,119 @@
+// This script can be run against a mainnet fork by spinning a node in a separate terminal with:
+// HARDHAT_CHAIN_ID=1 yarn hardhat node --fork https://mainnet.infura.io/v3/<YOUR-INFURA-KEY> --port 9545 --no-deploy
+// and then running this script with:
+// GCKMS_WALLET=<OPTIONAL-GCKMS-WALLET> \
+// yarn hardhat run packages/scripts/src/admin-proposals/register-parent-messenger/0_Propose.ts --network <network>
+
+const hre = require("hardhat");
+
+import { GovernorHubEthers, OracleHubEthers, ProposerV2Ethers, VotingTokenEthers } from "@uma/contracts-node";
+
+import { Provider } from "@ethersproject/abstract-provider";
+import { getGckmsSigner } from "@uma/common";
+import { OptimismParentMessenger } from "@uma/contracts-node/typechain/core/ethers";
+import { BigNumberish, Signer, Wallet } from "ethers";
+import { BytesLike } from "ethers/lib/utils";
+import { getAddress } from "../../upgrade-tests/register-new-contract/common";
+import { getContractInstance } from "../../utils/contracts";
+import { BASE_CHAIN_ID, PARENT_MESSENGER_DEFAULT_GAS_LIMIT } from "./common";
+
+interface AdminProposalTransaction {
+  to: string;
+  value: BigNumberish;
+  data: BytesLike;
+}
+
+const proposerWallet = "0x2bAaA41d155ad8a4126184950B31F50A1513cE25";
+
+async function main() {
+  const adminProposalTransactions: AdminProposalTransaction[] = [];
+
+  let proposerSigner: Signer;
+
+  if (process.env.GCKMS_WALLET) {
+    proposerSigner = ((await getGckmsSigner()) as Wallet).connect(hre.ethers.provider as Provider);
+    if (proposerWallet.toLowerCase() != (await proposerSigner.getAddress()).toLowerCase())
+      throw new Error("GCKMS wallet does not match proposer wallet");
+  } else {
+    proposerSigner = (await hre.ethers.getSigner(proposerWallet)) as Signer;
+  }
+
+  console.log("1. LOADING DEPLOYED CONTRACT STATE");
+
+  const votingToken = await getContractInstance<VotingTokenEthers>("VotingToken");
+
+  const proposer = await await getContractInstance<ProposerV2Ethers>("ProposerV2");
+
+  const baseParentMessengerAddress = await getAddress("Base_ParentMessenger", 1);
```

Comment:
Minor question: can we generalize this script to any OP stack chain? Is there anything here that's specific to Base? Or could we just pass in/look up the chain name given the chainId?

Reply:
Yes, I decided to hardcode for Base here because it's a custom script that also updates the defaultGasLimit, something we wouldn't do every time we register a new Parent Messenger. I will merge and run this governance proposal, and then we can remove those extra steps and generalize it if that makes sense.